static/src/javascripts/projects/commercial/modules/third-party-tags/outbrain-codes-new.js

```diff
@@ -0,0 +1,76 @@
+// @flow
+
+/*
+  Implements outbrain codes according to the new structure defined
+  here: https://trello.com/c/r1TDqVKP
+*/
+
+const normalizedTypes = {
+    defaults: 'defaults',
+    merchandising: 'nonCompliant',
+    nonCompliant: 'nonCompliant',
+};
+
+const outbrainCodes = {
+    defaults: {
+        desktop: { code: 'AR_12' },
+        tablet: { code: 'MB_6' },
+        mobile: { code: 'MB_4' },
+    },
+    epic: {
+        desktop: { code: 'AR_13' },
+        tablet: { code: 'MB_7' },
+        mobile: { code: 'MB_5' },
+    },
+    nonCompliant: {
+        desktop: { code: 'AR_14' },
+        tablet: { code: 'MB_8' },
+        mobile: { code: 'MB_10' },
+    },
+};
+
+const outbrainCode = function(
+    paramSet: string,
+    breakpoint: string
+): { code?: string, image?: string, text?: string } {
+    const byParamSet = outbrainCodes[paramSet];
+    if (!byParamSet) {
+        throw new Error(`Unknown outbrain param set  (${paramSet})`);
+    }
+    const byBreakpoint = byParamSet[breakpoint];
+    if (!byBreakpoint) {
+        throw new Error(`Unknown outbrain breakpoint (${breakpoint})`);
+    }
+    return { code: byBreakpoint.code };
+};
+
+const getCode = function(args: {
+    outbrainType: string,
+    contributionEpicVisible: boolean,
+    breakpoint: string,
+    section: string,
+}): { code?: string, image?: string, text?: string } {
+    console.log('OUTBRAIN -NEW getCode arguments', args);
```

Comment:
Is this meant to be here?

Reply:
Yes, as it is behind a 0% participation test. I left them in on purpose to help Outbrain see what code is going to be sent and from what it is built.